Nix build fails on aarch64-darwin

Title is self-explanatory, I was asked to create this issue along with a minimal nix flake to reproduce the issue.

@ncfavier I wonder if this is an issue specific to aarch64-darwin (assuming this is not your platform). This is, I believe, the salient message from the build log to begin investigation upon:

`"The following functions have been omitted by a nixpkgs-specific patch to Cabal: getLibDir, getDynLibDir, getLibexecDir, getSysconfDir"` used in the `Agda.Version` module.

flake.nix:
```nix
{
  description = "Test flake for Agda overlay";

  inputs = {
    nixpkgs.url = "github:NixOS/nixpkgs/nixos-unstable";
    agda.url = "github:agda/agda";
    agda.inputs.nixpkgs.follows = "nixpkgs";
  };

  outputs =
    {
      self,
      nixpkgs,
      agda,
    }:
    let
      system = "aarch64-darwin";
      pkgs = import nixpkgs {
        inherit system;
        overlays = [ agda.overlays.default ];
      };
    in
    {
      packages.${system}.default = pkgs.agda;

      devShells.${system}.default = pkgs.mkShell {
        buildInputs = [ pkgs.agda ];
      };
    };
}
```

And here's the build log:

```
$  nix build                                                                16
error: builder for '/nix/store/jdwbm8k38j9jm2ppdw8m85d1k01na49b-Agda-2.9.0.drv' failed with exit code 1;
       last 25 log lines:
       > Using strip found on system at:
       > /nix/store/cdjiamwcmyzbd48jz4nqdmxfpinp9d6w-clang-wrapper-19.1.7/bin/strip
       > Using tar found on system at:
       > /nix/store/25v7jy2zfx4mx2ca8grk8lkmyn8zbkap-gnutar-1.35/bin/tar
       > No uhc found
       > Running phase: buildPhase
       > Preprocessing executable 'agda-mode' for Agda-2.9.0..
       > Building executable 'agda-mode' for Agda-2.9.0..
       > [1 of 7] Compiling Agda.Setup.DataFiles ( src/setup/Agda/Setup/DataFiles.hs, dist/build/agda-mode/agda-mode-tmp/Agda/Setup/DataFiles.o, dist/build/agda-mode/agda-mode-tmp/Agda/Setup/DataFiles.dyn_o )
       > [2 of 7] Compiling Paths_Agda       ( dist/build/agda-mode/autogen/Paths_Agda.hs, dist/build/agda-mode/agda-mode-tmp/Paths_Agda.o, dist/build/agda-mode/agda-mode-tmp/Paths_Agda.dyn_o )
       > [3 of 7] Compiling Agda.Version     ( src/setup/Agda/Version.hs, dist/build/agda-mode/agda-mode-tmp/Agda/Version.o, dist/build/agda-mode/agda-mode-tmp/Agda/Version.dyn_o )
       >
       > src/setup/Agda/Version.hs:13:1: error: [GHC-15328] [-Wdeprecations, -Werror=deprecations]
       >     Module ‘Paths_Agda’:
       >       "The following functions have been omitted by a nixpkgs-specific patch to Cabal: getLibDir, getDynLibDir, getLibexecDir, getSysconfDir"
       >    |
       > 13 | import qualified Paths_Agda as PA
       >    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
       >
       > src/setup/Agda/Version.hs:19:27: error: [GHC-63394] [-Wdeprecations, -Werror=deprecations]
       >     In the use of ‘version’ (imported from Paths_Agda):
       >     "The following functions have been omitted by a nixpkgs-specific patch to Cabal: getLibDir, getDynLibDir, getLibexecDir, getSysconfDir"
       >    |
       > 19 |             versionBranch PA.version
       >    |                           ^^^^^^^^^^
       For full logs, run 'nix log /nix/store/jdwbm8k38j9jm2ppdw8m85d1k01na49b-Agda-2.9.0.drv'.
error: 1 dependencies of derivation '/nix/store/4gphy2bxkh13nr4b5phnl00yzb2sl3kz-agdaWithPackages-2.9.0.drv' failed to build
```

--- a/flake.nix
+++ b/flake.nix
@@ -35,7 +35,7 @@
       # Various builds of Agda
 
       # Recommended build
-      agda-pkg = hlib.overrideCabal (_: {
+      agda-pkg = hlib.overrideCabal (drv: {
           # These settings are documented at
           # https://nixos.org/manual/nixpkgs/unstable/#haskell-mkderivation
 
@@ -53,6 +53,13 @@
           # Place the binaries in a separate output with a much smaller closure size.
           enableSeparateBinOutput = true;
           mainProgram = "agda";
+        } // pkgs.lib.optionalAttrs (pkgs.stdenv.hostPlatform.isDarwin && pkgs.stdenv.hostPlatform.isAarch64) {
+          # A nixpkgs-specific patch for aarch64-darwin related to the separate bin output
+          # causes a warning about some functions being removed from Paths_Agda, which
+          # we can just ignore. See https://github.com/agda/agda/issues/8016
+          configureFlags = drv.configureFlags or [] ++ [
+            "--ghc-option=-Wwarn=deprecations"
+          ];
         }) agda-pkg-minimal;
 
       # An even faster Agda build, achieved by asking GHC to optimize less

